added app engine smoke test

danielpeach opened the pull request:
@ewiseblatt please review.

ewiseblatt commented on testing/citest/spinnaker_testing/spinnaker_test_scenario.py:
```diff
@@ -318,6 +333,12 @@ def _initObservationConfigurationParameters(cls, parser, defaults):
         help='aws command-line tool --profile parameter when observing AWS.')
+    
+    parser.add_argument(
+        '--appengine_credentials_path',
```
Mention the default here is to use --gce_credentials_path whose default is application credentials.

I'm not sure how you'd use application credentials for appengine but an explicit service account for GCE.

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,314 @@
+
+Sample Usage:
+    Assuming you have created $PASSPHRASE_FILE (which you should chmod 400)
+    and $CITEST_ROOT points to the root directory of this repository
```
you can use ssh-agent in the shell instead of the passphrase file.

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,314 @@
+    self.__gcp_project = bindings['APPENGINE_PRIMARY_MANAGED_PROJECT_ID']
+    self.__cluster_name = frigga.Naming.cluster(self.TEST_APP, self.TEST_STACK)
+    self.__server_group_name = frigga.Naming.server_group(self.TEST_APP, self.TEST_STACK)
+    self.__lb_name = 'citest'
```
If this is under your control, dont use a constant name. If it is derived, dont have it derive constant names. You want multiple tests to be able to run without necessarily conflicting with one another. If one test leaks the load balancer you dont want that to influence another run.

The google_smoke_test does this:
   self.__lb_name = '{app}-{stack}-{detail}'.format(
	app=bindings['TEST_APP'], stack=bindings['TEST_STACK'],
        detail=self.__lb_detail)

It should be updated to use frigga as you are elsewhere already.

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,314 @@
+    self.__cluster_name = frigga.Naming.cluster(self.TEST_APP, self.TEST_STACK)
+    self.__server_group_name = frigga.Naming.server_group(self.TEST_APP, self.TEST_STACK)
+    self.__lb_name = 'citest'
+    self.__repo_url = 'https://github.com/danielpeach/python-docs-samples.git'
```
Not sure what this is for, are you sure you want to reference your repo to anonymous people that may run this and copy and paste future things from it?

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,314 @@
+    self.pipeline_id = None
+    
+  def create_app(self):
+    contract = jc.Contract()
```
mention you arent testing create/delete app because they are well covered elsewhere so assume they work.

danielpeach commented:
I updated this. I've added flags to pass in the hard-coded values you were concerned with, though `lb_name` can't be dynamic, since it's determined by the App Engine application and not Spinnaker. 

However, just by the nature of how load balancers work in App Engine, there should be no issue between tests if there is a resource leak (or at least there is no danger of a name collision).

ewiseblatt commented:
What is the App Engine application? Is that this test? If so, then this
test could determine the name. Bake the test_id into the name, or like
everyone else, the APP_NAME (and STACK_NAME). In turn those bake the
test_id. At least this way you correlated the LB instance with the test
instance that created it.

On Fri, Feb 17, 2017 at 3:13 PM, Daniel Peach <notifications@github.com>
wrote:

> I updated this. I've added flags to pass in the hard-coded values you were
> concerned with, though lb_name can't be dynamic, since it's determined by
> the App Engine application and not Spinnaker.
>
> However, just by the nature of how load balancers work in App Engine,
> there should be no issue between tests if there is a resource leak (or at
> least there is no danger of a name collision).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/spinnaker/pull/1433#issuecomment-280754500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEKkGNj350YB9mcZ63C7HezqkyJ8mXGgks5rdf9QgaJpZM4MDgEK>
> .
>

danielpeach commented:
What you're asking will be possible after https://github.com/spinnaker/clouddriver/pull/1461 has been merged.

danielpeach commented:
@ewiseblatt updated.

ewiseblatt commented on testing/citest/spinnaker_testing/spinnaker_test_scenario.py:
```diff
@@ -318,6 +333,13 @@ def _initObservationConfigurationParameters(cls, parser, defaults):
+        default=defaults.get('APPENGINE_CREDENTIALS_PATH', None),
+        help='A path to the JSON file with credentials to use for observing'
+             ' tests run against Google App Engine. Defaults to the value set for'
+             '--gce_credentials_path, which defaults to application credentials.')
```
application default credentials

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,339 @@
+    super(AppengineSmokeTestScenario, cls).initArgumentParser(
+          parser, defaults=defaults)
+    parser.add_argument('--source_repo_url', default=None, 
+                        help='URL of a source code repository used by Spinnaker to deploy to App Engine.')
```
Are these required? Can the test just run without this ?

danielpeach replied:
Yes - this argument and the one you've pointed out below are required. The test now throws an error if you haven't supplied these.

ewiseblatt commented on testing/citest/tests/appengine_smoke_test.py:
```diff
@@ -0,0 +1,339 @@
+                                  '   static_dir: .']).format(service=self.__lb_name))
+    
+    self.__repo_url = bindings['SOURCE_REPO_URL']
+    self.__app_directory_root = bindings['APP_DIRECTORY_ROOT']
```
is it ok these are empty by default
